giskard/client/giskard_client.py

```diff
@@ -74,6 +73,33 @@ def __call__(self, r):
         return r
 
 
+def _cut_str_to_real_char(s, max_chars):
+    result = ""
+    length = 0
+
+    for char in s:
+        b = bytes(char, "utf-8")
+        length += len(b) // 2 + (len(b) % 2 > 0)
```

Comment:
Do you have a reference for this? UTF-8 could have 1-4 bytes. From my understanding of the code:
- for 1 byte (generally ASCII): `length += 0 + 1`
- for 2 bytes: `length += 1 + 0`
- for 3 bytes: `length += 1 + 1`
- for 4 bytes: `length += 2 + 0`

How do we determine the total length?

Reply:
Well Python count surrogate pairs has one character while Java, js, h2 and postgres count each individual character: https://en.wikipedia.org/wiki/UTF-16 

I improved the code as it wasn't 100% correct